lib/cosmos/packets/parsers/packet_item_parser.rb

```diff
@@ -93,7 +97,15 @@ def get_bit_size
     def get_array_size
       return nil unless (@parser.keyword.include?('ARRAY'))
       index = append? ? 3 : 4
-      Integer(@parser.parameters[index])
+      array_bit_size = Integer(@parser.parameters[index])
+      items = array_bit_size / get_bit_size()
+      if items >= BIG_ARRAY_SIZE
+        Logger.warn("In #{@parser.filename}:#{@parser.line_number} your definition of:")
+        Logger.warn(@parser.line)
+        Logger.warn("creates an array with #{items} elements. This may impact performance! "\
+          "Consider creating a BLOCK if this is binary data.")
```

Comment:
The only bummer about this is that it only appears the first time the cmd/tlm definitions are parsed

Reply:
Add to @warnings and it will print every time.